File: examples/getGPIO.c
#include <stdio.h>
#include <string.h>
#include <time.h>

#include <pi-gpio.h>
/*
List GPIO, Value, Function, Pullup/down (Pi4 only) for GPIO 0-27
*/

int main() {
  int n, f, p;
  setup();

  for (int i = 0; i < 28; i++) {
    n = input_gpio(i);
    f = gpio_function(i);
    p = (f == 0) ? get_pullupdn(i) : 0; // only input
    if (f == 0)
      printf("GPIO %2i Value=%i Function %i Pull %i\n", i, n, f, p);
    else
      printf("GPIO %2i Value=%i Function %i\n", i, n, f);
  }

  cleanup();
  return 0;
}

File: examples/getGPIO.py
#! /usr/bin/env python3
# 2022-11-25

import pi_gpio as GPIO

GPIO.setup()

print('GPIO {:08x}'.format(GPIO.input_28()))
for g in range(0, 28):
    v = GPIO.input_gpio(g)
    if(v):
        print("GPIO {:2d} Value={} Function {}".format(g, v, GPIO.gpio_function(g),))
    else:
        print("GPIO {:2d} Value={} Function {} Pull {}".format(g, v, GPIO.gpio_function(g), GPIO.get_pullupdn(g)))

GPIO.cleanup()
